src/systems/filecoin_vm/sysactors/init_actor.go

```diff
@@ -21,14 +21,11 @@ type Runtime = vmr.Runtime
 type Bytes = util.Bytes
 type Serialization = util.Serialization
 
-func (a *InitActorCode_I) State(rt Runtime) (vmr.ActorStateHandle, InitActorState) {
+func _loadState(rt Runtime) (vmr.ActorStateHandle, InitActorState) {
 	h := rt.AcquireState()
 	stateCID := ipld.CID(h.Take())
 	if ipld.CID_Equals(stateCID, ipld.EmptyCID()) {
-		if rt.CurrMethodNum() != actor.MethodConstructor {
-			rt.Abort("Actor state not initialized")
-		}
-		return h, nil
+		rt.Abort("Actor state not initialized")
```

Comment:
Why is CurrMethodNum (and this logic) no longer needed? This was originally added because, prior to their Constructor method being called, actors have an empty state CID by default (which does not deserialize properly into a value of the desired substate struct type).

Reply:
This method is no longer called by the constructor, which ignored the return value anyway. Hence the predicate that was on line 28 would always be satisfied.